Tribler/community/tunnel/tunnel_community.py

```diff
@@ -1212,13 +1212,13 @@ def crypto_out(self, circuit_id, content, is_data=False):
                                            *self.get_session_keys(self.relay_session_keys[circuit_id], ORIGINATOR))
         raise CryptoException("Don't know how to encrypt outgoing message for circuit_id %d" % circuit_id)
 
-    def crypto_in(self, circuit_id, content, is_data=False):
-        circuit = self.circuits.get(circuit_id, None)
+    def crypto_in(self, circuit, content, is_data=False):
+        circuit_id = circuit.circuit_id
```

Comment:
Interesting, so the id is inside the relay_session_key dict, what does that mean? Also thanks for taking a look at my PR! I really appreciate the feedback on the TODO's and changes.